### ISSUE TITLE
Build on MacOS

### ISSUE THREAD

**Issue by shiba24** [ImperialCollegeLondon/EventEMin#1]
Hi authors, thank you for providing such great code!

I have a question about your environment. I trying to build on my Mac. I understand you haven't tested it and building on Mac is sometimes not straightforward than Ubuntu.
But let me try it ... I am facing some (probably version?) error in build. So basically I'd like to know which version you are actually using.

# Environment

I have: 

- cmake 3.21.3
- Eigen 3.4.0_1
- gsl 2.7

# Error

## tl;dr

The error that I'm facing is two types. The first one is 

```
EventEMin/include/dispersion/approx_entropy/approx_entropy.h:116:5: error: no matching function for call to 'gaussKernel'
```

And the second one is

```
EventEMin/include/dispersion/approx_entropy/approx_entropy.h:171:33: error: non-type template argument is not a constant expression
```

I took a look into the second one and found [this](https://stackoverflow.com/questions/34561884/is-this-error-message-correct-non-type-template-argument-is-not-a-constant-expr/34675596) and [this](https://stackoverflow.com/questions/24580714/why-is-this-not-a-constant-expression). And that's why I guessing some version problem - especially around C++.

Could you provide your tested environment version information if you can?

## The full error log:

```

Consolidate compiler generated dependencies of target example_approx_entropy_homography
[ 43%] Building CXX object test/CMakeFiles/example_approx_entropy_homography.dir/main_approx_entropy_homography.cpp.o
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_potential.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/convolution.h:7:
In file included from /usr/local/include/eigen3/Eigen/Core:176:
/usr/local/include/eigen3/Eigen/src/Core/arch/Default/Half.h:537:9: warning: compound literals are a C99-specific feature [-Wc99-extensions]
  h.x = _cvtss_sh(ff, 0);
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/include/f16cintrin.h:69:54: note: expanded from macro '_cvtss_sh'
  (unsigned short)(((__v8hi)__builtin_ia32_vcvtps2ph((__v4sf){a, 0, 0, 0}, \
                                                     ^
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_potential.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/convolution.h:8:
In file included from /usr/local/include/eigen3/unsupported/Eigen/CXX11/Tensor:69:
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIndexList.h:44:36: warning: unused parameter 'val' [-Wunused-parameter]
  EIGEN_DEVICE_FUNC void set(Index val) {
                                   ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIndexList.h:60:54: warning: unused parameter 'val' [-Wunused-parameter]
  EIGEN_DEVICE_FUNC void set(const IndexPair<Index>& val) {
                                                     ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorIndexList.h:276:49: warning: unused parameter 'i' [-Wunused-parameter]
  EIGEN_DEVICE_FUNC static void set(const Index i, IndexTuple<T...>& t, const ValueT value) {
                                                ^
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_potential.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/convolution.h:8:
In file included from /usr/local/include/eigen3/unsupported/Eigen/CXX11/Tensor:90:
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorContraction.h:255:46: warning: unused parameter 'beta' [-Wunused-parameter]
      const ResScalar alpha, const ResScalar beta) {
                                             ^
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:10:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/entropy/potential.h:4:
/Users/shintaro/work/tub/EventEMin/include/dispersion/entropy/entropy.h:43:15: warning: unused parameter 'd' [-Wunused-parameter]
    const int d) const
              ^
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_potential.h:4:
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:116:5: error: no matching function for call to 'gaussKernel'
    kernel::gaussKernel<T, this->ndims()>(ksize, kernel_);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_tsallis.h:45:5: note: in instantiation of member function 'event_model::dispersion::ApproxEntropy<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::ApproxEntropy' requested here
    ApproxEntropy<ApproxTsallis<Model> >(dim, offset, lambda, ksize),
    ^
/Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:68:14: note: in instantiation of member function 'event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>::ApproxTsallis' requested here
  Dispersion dispersion({width, height});
             ^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:163:1: note: candidate template ignored: substitution failure: too many template arguments for function template 'gaussKernel'
gaussKernel(
^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:179:1: note: candidate template ignored: substitution failure: too many template arguments for function template 'gaussKernel'
gaussKernel(
^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:147:1: note: candidate template ignored: invalid explicitly-specified argument for template parameter 'NDims'
gaussKernel(
^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:222:1: note: candidate function template not viable: requires 4 arguments, but 2 were provided
gaussKernel(
^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:242:1: note: candidate function template not viable: requires 4 arguments, but 2 were provided
gaussKernel(
^
/Users/shintaro/work/tub/EventEMin/include/gauss_kernel.h:260:1: note: candidate function template not viable: requires 4 arguments, but 2 were provided
gaussKernel(
^
In file included from /Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion.h:4:
In file included from /Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_potential.h:4:
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:171:33: error: non-type template argument is not a constant expression
    convolution::Convolution<U, this->ndims(), T> conv(dim(),
                                ^~~~~~~~~~~~~
/Users/shintaro/work/tub/EventEMin/include/dispersion/dispersion.h:170:32: note: in instantiation of function template specialization 'event_model::dispersion::ApproxEntropy<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::compute<float>' requested here
    (*f)(0)=this->underlying().compute(cmScaled);
                               ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/../src/AutoDiff/AutoDiffJacobian.h:73:16: note: in instantiation of function template specialization 'event_model::dispersion::Dispersion<event_model::dispersion::ApproxEntropy<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>>::operator()<float>' requested here
      Functor::operator()(x, v, Params...);
               ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:210:5: note: in instantiation of function template specialization 'Eigen::AutoDiffJacobian<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::operator()<>' requested here
    func_(vars, &f, nullptr);
    ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:254:10: note: in instantiation of member function 'event_model::optimiser::GSLfdfOptimiser<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::operator()' requested here
  return (*static_cast<GSLfdfOptimiser<Func>*>(params))(gvars);
         ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:92:17: note: in instantiation of function template specialization 'event_model::optimiser::gslfdffFunc<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>' requested here
    gslFunc_.f=&gslfdffFunc<Func>;
                ^
/Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:84:13: note: in instantiation of member function 'event_model::optimiser::GSLfdfOptimiser<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::GSLfdfOptimiser' requested here
  Optimiser optimiser(dispersion, Optimiser::OptimiserParams(
            ^
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:171:33: note: use of 'this' pointer is only allowed within the evaluation of a call to a 'constexpr' member function
    convolution::Convolution<U, this->ndims(), T> conv(dim(),
                                ^
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:171:33: error: non-type template argument is not a constant expression
    convolution::Convolution<U, this->ndims(), T> conv(dim(),
                                ^~~~~~~~~~~~~
/Users/shintaro/work/tub/EventEMin/include/dispersion/dispersion.h:170:32: note: in instantiation of function template specialization 'event_model::dispersion::ApproxEntropy<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::compute<Eigen::AutoDiffScalar<Eigen::Matrix<float, 8, 1, 0>>>' requested here
    (*f)(0)=this->underlying().compute(cmScaled);
                               ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/../src/AutoDiff/AutoDiffJacobian.h:93:14: note: in instantiation of function template specialization 'event_model::dispersion::Dispersion<event_model::dispersion::ApproxEntropy<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>>::operator()<Eigen::AutoDiffScalar<Eigen::Matrix<float, 8, 1, 0>>>' requested here
    Functor::operator()(ax, &av, Params...);
             ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:210:5: note: in instantiation of function template specialization 'Eigen::AutoDiffJacobian<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::operator()<>' requested here
    func_(vars, &f, nullptr);
    ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:254:10: note: in instantiation of member function 'event_model::optimiser::GSLfdfOptimiser<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::operator()' requested here
  return (*static_cast<GSLfdfOptimiser<Func>*>(params))(gvars);
         ^
/Users/shintaro/work/tub/EventEMin/include/optimiser/gsl_fdf_optimiser.h:92:17: note: in instantiation of function template specialization 'event_model::optimiser::gslfdffFunc<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>' requested here
    gslFunc_.f=&gslfdffFunc<Func>;
                ^
/Users/shintaro/work/tub/EventEMin/test/main_approx_entropy_homography.cpp:84:13: note: in instantiation of member function 'event_model::optimiser::GSLfdfOptimiser<event_model::dispersion::ApproxTsallis<event_model::Model<event_model::model::homography::Homography<float>>>>::GSLfdfOptimiser' requested here
  Optimiser optimiser(dispersion, Optimiser::OptimiserParams(
            ^
/Users/shintaro/work/tub/EventEMin/include/dispersion/approx_entropy/approx_entropy.h:171:33: note: use of 'this' pointer is only allowed within the evaluation of a call to a 'constexpr' member function
    convolution::Convolution<U, this->ndims(), T> conv(dim(),
                                ^
6 warnings and 3 errors generated.
make[2]: *** [test/CMakeFiles/example_approx_entropy_homography.dir/main_approx_entropy_homography.cpp.o] Error 1
make[1]: *** [test/CMakeFiles/example_approx_entropy_homography.dir/all] Error 2

```

**Comment by shiba24** [comment]
Thank you for your quick reply! I tried it, the previous errors resolved 🎉 , and the error has changed now to:

# tldr (only error lines);

- `EventEMin/include/dispersion/dispersion.h:49:3: error: call to implicitly-deleted default constructor of 'const event_model::DispersionBase<event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>>::Model' (aka 'const event_model::Model<event_model::model::Rotation<float>>')`

- `EventEMin/include/gauss_kernel.h:81:10: error: no matching member function for call to 'resize'`

- `EventEMin/include/gauss_kernel.h:87:22: error: no matching function for call to object of type 'Tensor<float, 2>'`

- `EventEMin/include/gauss_kernel.h:65:5: error: no matching function for call to object of type 'Tensor<float, 2>'`



# Full error

```
[ 10%] Building CXX object test/CMakeFiles/example_test_sequence.dir/main_test_sequence.cpp.o
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/potential.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/convolution.h:9:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/types_def.h:4:
In file included from /usr/local/include/eigen3/Eigen/Core:176:
/usr/local/include/eigen3/Eigen/src/Core/arch/Default/Half.h:537:9: warning: compound literals are a C99-specific feature [-Wc99-extensions]
  h.x = _cvtss_sh(ff, 0);
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/include/f16cintrin.h:69:54: note: expanded from macro '_cvtss_sh'
  (unsigned short)(((__v8hi)__builtin_ia32_vcvtps2ph((__v4sf){a, 0, 0, 0}, \
                                                     ^
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/potential.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:5:
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/dispersion.h:49:3: error: call to implicitly-deleted default constructor of 'const event_model::DispersionBase<event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>>::Model' (aka 'const event_model::Model<event_model::model::Rotation<float>>')
  DispersionBase(void)
  ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:36:9: note: in instantiation of member function 'event_model::DispersionBase<event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>>::DispersionBase' requested here
      : DispersionBase<ApproxDispersion<Derived> >(),
        ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/tsallis.h:39:9: note: in instantiation of member function 'event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>::ApproxDispersion' requested here
      : ApproxDispersion<ApproxTsallis<Model> >(dimScaleMax, offset, lambda,
        ^
/Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:96:14: note: in instantiation of member function 'event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>::ApproxTsallis' requested here
  Dispersion dispersion(width);
             ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/model/model.h:35:3: note: explicitly defaulted function was implicitly deleted here
  Model(void) = default;
  ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/model/model.h:32:14: note: default constructor of 'Model<event_model::model::Rotation<float>>' is implicitly deleted because field 'func_' of const-qualified type 'const event_model::Model<event_model::model::Rotation<float>>::Func' (aka 'const event_model::model::Rotation<float>') would not be initialized
  const Func func_;
             ^
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/potential.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/dispersion.h:8:
/Users/shintaro/work/tub/paper-code/EventEMin/include/gauss_kernel.h:81:10: error: no matching member function for call to 'resize'
  kernel.resize(sizeArray);
  ~~~~~~~^~~~~~
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:53:13: note: in instantiation of function template specialization 'event_model::kernel::gaussKernel<float, 2>' requested here
    kernel::gaussKernel<T, NDims>(ksize, kernel_);
            ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/tsallis.h:39:9: note: in instantiation of member function 'event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>::ApproxDispersion' requested here
      : ApproxDispersion<ApproxTsallis<Model> >(dimScaleMax, offset, lambda,
        ^
/Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:96:14: note: in instantiation of member function 'event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>::ApproxTsallis' requested here
  Dispersion dispersion(width);
             ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:438:10: note: candidate function template not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'Eigen::Tensor<float, 2, 0>::Index' (aka 'long') for 1st argument
    void resize(Index firstDimension, IndexTypes... otherDimensions)
         ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:447:28: note: candidate function not viable: no known conversion from 'array<int, 2>' to 'const array<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' for 1st argument
    EIGEN_DEVICE_FUNC void resize(const array<Index, NumIndices>& dimensions)
                           ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:465:28: note: candidate function not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'const DSizes<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' (aka 'const DSizes<long, NumIndices>') for 1st argument
    EIGEN_DEVICE_FUNC void resize(const DSizes<Index, NumIndices>& dimensions) {
                           ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:483:10: note: candidate template ignored: could not match 'IndexList' against 'array'
    void resize(const Eigen::IndexList<FirstType, OtherTypes...>& dimensions) {
         ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:506:10: note: candidate template ignored: could not match 'Sizes' against 'array'
    void resize(const Sizes<Indices...>& dimensions) {
         ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:474:10: note: candidate function not viable: requires 0 arguments, but 1 was provided
    void resize()
         ^
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/potential.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/dispersion.h:8:
/Users/shintaro/work/tub/paper-code/EventEMin/include/gauss_kernel.h:87:22: error: no matching function for call to object of type 'Tensor<float, 2>'
  kernel = kernel / (kernel(sizeArray) * std::pow(static_cast<T>(2.0 * M_PI),
                     ^~~~~~
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:292:51: note: candidate function not viable: no known conversion from 'array<int, 2>' to 'const array<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()(const array<Index, NumIndices>& indices)
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:308:51: note: candidate function not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'Eigen::Tensor<float, 2, 0>::Index' (aka 'long') for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()(Index index)
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:236:57: note: candidate function not viable: no known conversion from 'array<int, 2>' to 'const array<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()(const array<Index, NumIndices>& indices) const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:241:57: note: candidate function not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'Eigen::Tensor<float, 2, 0>::Index' (aka 'long') for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()(Index index) const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:247:57: note: candidate function not viable: requires 0 arguments, but 1 was provided
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()() const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:314:51: note: candidate function not viable: requires 0 arguments, but 1 was provided
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()()
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:195:26: note: candidate function template not viable: requires at least 2 arguments, but 1 was provided
    inline const Scalar& operator()(Index firstIndex, Index secondIndex, IndexTypes... otherIndices) const
                         ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:262:20: note: candidate function template not viable: requires at least 2 arguments, but 1 was provided
    inline Scalar& operator()(Index firstIndex, Index secondIndex, IndexTypes... otherIndices)
                   ^
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/potential.h:4:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:5:
In file included from /Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/dispersion.h:8:
/Users/shintaro/work/tub/paper-code/EventEMin/include/gauss_kernel.h:65:5: error: no matching function for call to object of type 'Tensor<float, 2>'
    kernel(ind) = gauss<T>(p);
    ^~~~~~
/Users/shintaro/work/tub/paper-code/EventEMin/include/gauss_kernel.h:85:3: note: in instantiation of function template specialization 'event_model::kernel::gaussKernelIterate<float, 2>' requested here
  gaussKernelIterate<T, N>(ksize, hksize, 0, p, ind, kernel);
  ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/dispersion.h:53:13: note: in instantiation of function template specialization 'event_model::kernel::gaussKernel<float, 2>' requested here
    kernel::gaussKernel<T, NDims>(ksize, kernel_);
            ^
/Users/shintaro/work/tub/paper-code/EventEMin/include/dispersion/approx_dispersion/tsallis.h:39:9: note: in instantiation of member function 'event_model::ApproxDispersion<event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>>::ApproxDispersion' requested here
      : ApproxDispersion<ApproxTsallis<Model> >(dimScaleMax, offset, lambda,
        ^
/Users/shintaro/work/tub/paper-code/EventEMin/test/main_test_sequence.cpp:96:14: note: in instantiation of member function 'event_model::ApproxTsallis<event_model::Model<event_model::model::Rotation<float>>>::ApproxTsallis' requested here
  Dispersion dispersion(width);
             ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:292:51: note: candidate function not viable: no known conversion from 'array<int, 2>' to 'const array<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()(const array<Index, NumIndices>& indices)
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:308:51: note: candidate function not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'Eigen::Tensor<float, 2, 0>::Index' (aka 'long') for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()(Index index)
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:236:57: note: candidate function not viable: no known conversion from 'array<int, 2>' to 'const array<Eigen::Tensor<float, 2, 0>::Index, NumIndices>' for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()(const array<Index, NumIndices>& indices) const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:241:57: note: candidate function not viable: no known conversion from 'Array<int, 2>' (aka 'array<int, 2UL>') to 'Eigen::Tensor<float, 2, 0>::Index' (aka 'long') for 1st argument
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()(Index index) const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:247:57: note: candidate function not viable: requires 0 arguments, but 1 was provided
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE const Scalar& operator()() const
                                                        ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:314:51: note: candidate function not viable: requires 0 arguments, but 1 was provided
    EIGEN_DEVICE_FUNC EIGEN_STRONG_INLINE Scalar& operator()()
                                                  ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:195:26: note: candidate function template not viable: requires at least 2 arguments, but 1 was provided
    inline const Scalar& operator()(Index firstIndex, Index secondIndex, IndexTypes... otherIndices) const
                         ^
/usr/local/include/eigen3/unsupported/Eigen/CXX11/src/Tensor/Tensor.h:262:20: note: candidate function template not viable: requires at least 2 arguments, but 1 was provided
    inline Scalar& operator()(Index firstIndex, Index secondIndex, IndexTypes... otherIndices)
```

**Comment by varun19299** [comment]
Does this work with OpenMP enabled? I was able to compile while turning OpenMP off: I'm aware that openMP doesn't play well with Apple Clang.
